File: Program23.c
#include<stdio.h>

void DisplayOddFactors(int iNo){

    int iCnt=0;

    for(iCnt=1;iCnt<=iNo;iCnt++){
        if((iNo%iCnt)==0){
                if(iCnt%2!=0){
                    printf("%d\n",iCnt);
                } 
        }
    }
}
int main(){

    int iValue=0;

    printf("enter the number:");
    scanf("%d",&iValue);

    DisplayOddFactors(iValue);

    return 0;
}

File: Program23.java
//Write a Java program to count the number of characters in a string.
import java.util.*;

class StringX{
    public int CountLetters(String s){
        int iCnt=0,iCount=0;
        for(iCnt=0;iCnt<s.length();iCnt++){
            iCount++;
        }

        return iCount;
    }
}
public class Program23 {
    public static void main(String arr[]){
        Scanner obj=new Scanner(System.in);

        System.out.println("enter string");
        String str=obj.nextLine();

        StringX sobj=new StringX();
        int iRet=sobj.CountLetters(str);

        System.out.println(iRet);

    }
}
